In [5]:
import pandas as pd

pertama kita akna mengimport library yang dibutuhkan, pada kasus ini adalah library pandas.

In [17]:
path = 'tae.data'

tidak lupa untuk mendeklarasikan pathnya agar terbaca.

In [18]:
names = ['EnglishSpeaker','CourseInstructor','Course','SummerOrRegular','ClassSize','ClassAttribute']

lalu dilanjut dengan penamaan setiap fiturnya.

In [19]:
dataset = pd.read_csv(path, names=names)

kemudian deklarasikan dataset agar bisa terbaca menggunakan function dari pandas.

In [20]:
print(dataset.shape)

(151, 6)


dan jumlah data beserta jumlah fiturnya akan di tampilkan dengan kode di atas.

In [21]:
print(dataset.head(20))

    EnglishSpeaker  CourseInstructor  Course  SummerOrRegular  ClassSize  \
0                1                23       3                1         19   
1                2                15       3                1         17   
2                1                23       3                2         49   
3                1                 5       2                2         33   
4                2                 7      11                2         55   
5                2                23       3                1         20   
6                2                 9       5                2         19   
7                2                10       3                2         27   
8                1                22       3                1         58   
9                2                15       3                1         20   
10               2                10      22                2          9   
11               2                13       1                2         30   
12          

untuk fitur "SummerOrRegular" 1 = Summer 2 = Regular
English Speaker = 1 = yes 2 = no
class atribute = 1 = low, 2 = Medium, 3 = High.

kemudain print data sebanyak 20 data pertama untuk melihat data yang kan kita olah.

In [22]:
dataset.shape

(151, 6)

In [23]:
from sklearn import model_selection

lalu kita emgnimport model seleksi dengan library sklearn

In [25]:
array = dataset.values

X = array[:,0:5]
Y = array[:,3]

validation_size = 0.20

seed = 7

X_train, X_validation, Y_train, Y_validation = model_selection.train_test_split(X, Y, test_size=validation_size,
random_state=seed)

kemudian kita mendeklarasikan array, dengan dataset.values dan deklrasikan pula konstanta X dan Y dimana X adalah untuk jumlah fitur yang ada, dan y adalah jumlah klasifikasi kelas yang ada, pada kasus ini terdapat 3 kelas, yaiut 1 = low 2 = medium, 3 = high. dan untuk seednya dideklrasikan menjadi 7, dan kemudian dari dataset kita akan dipisah dengan perbandingan 80 : 20, 80 berupa data training dan 20 berupa data testing

In [26]:
print("X_train :", X_train.shape)
print("X_validation :", X_validation.shape)
print("Y_train :", Y_train.shape)
print("Y_validation :", Y_validation.shape)

X_train : (120, 5)
X_validation : (31, 5)
Y_train : (120,)
Y_validation : (31,)


kode di atas untuk menampilkan hasil output dari masing2 data training dan juga validation, dimana 80% dari total datanya adalah 120, jadi terdapat 120 data training yang akan dipakai untuk melatih nmodel, dan 20% dari datasetnya adalah sebanyak 31 yang akan di pakai sebagai data testing atau validation.

In [27]:
seed = 7

scoring = 'accuracy'

lalu untuk scoringnya menggunakan "accuray" meskipun terdapat jenis scoring lain yang dapat digunakan seperti precission dan juga recall yang ke4mudian dipakai untuk f1 score.

In [30]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC

models = []

models.append(('KNN', KNeighborsClassifier()))
models.append(('NB', GaussianNB()))
models.append(('SVM', SVC()))


kemudian kita akna mengimport semua library yang diperlukan yang kemudian akan dibandingkan menggunakan meotde mana yang akan menghasilkan output maksimal.

In [31]:
results = []
names = []
for name, model in models:
  kfold = model_selection.KFold(n_splits=10, random_state=seed, shuffle=True)
  cv_results = model_selection.cross_val_score(model, X_train, Y_train,
  cv=kfold, scoring=scoring)
  results.append(cv_results)
  names.append(name)
  msg = "%s: %f (%f)" % (name, cv_results.mean(), cv_results.std())
  print(msg)

KNN: 0.900000 (0.097183)
NB: 1.000000 (0.000000)
SVM: 0.850000 (0.081650)


dari outputnya, kita dapat melihat bahwa hasil output maksimal terdapat pada Naive Bayes dengan sebesar 1.000

In [33]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [34]:
svm = SVC()

svm.fit(X_train, Y_train)
predictions = svm.predict(X_validation)
print(accuracy_score(Y_validation, predictions))

print(confusion_matrix(Y_validation, predictions))
print(classification_report(Y_validation, predictions))

0.8387096774193549
[[ 0  5]
 [ 0 26]]
              precision    recall  f1-score   support

           1       0.00      0.00      0.00         5
           2       0.84      1.00      0.91        26

    accuracy                           0.84        31
   macro avg       0.42      0.50      0.46        31
weighted avg       0.70      0.84      0.77        31



/usr/local/lib/python3.10/dist-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.10/dist-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.10/dist-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


hasil di atas kita dapat menyimpulkan bahwa total yang diprediksi benar dibandingkan dengan total data yang ada. dan ia memiliki confuson matrix sebesar 0 5 0 26. dan terakhir terdapat laporan klasifikasi dimana pertama terdapat precision yaitu proporsi positif yang benar atau true positif dibandingkan dengan total presiksi positif, kemudian ada recall dimana ia adalah proporsi positif yang dibandingkna dengan data yang sebenernya positif, dan f1 score adlah penggabungannya. intinya adalah data memiliki tingkat akurasi yang cukup tinggi yaitu sebesar 83.87%.